lib/routes/veeam/head.js

```diff
@@ -0,0 +1,44 @@
+const xml2js = require('xml2js');
+const { errors } = require('arsenal');
+const metadata = require('../../metadata/wrapper');
+const { getResponseHeader, buildHeadXML } = require('./utils');
+const { responseXMLBody, responseContentHeaders } = require('arsenal/build/lib/s3routes/routesUtils');
+
+/**
+ * Returns system.xml or capacity.xml files metadata for a given bucket.
+ *
+ * @param {object} request - request object
+ * @param {object} response - response object
+ * @param {object} bucketMd - bucket metadata from the db
+ * @param {object} log - logger object
+ * @returns {undefined} -
+ */
+function headVeeamFile(request, response, bucketMd, log) {
+    if (!bucketMd) {
+        return responseXMLBody(errors.NoSuchBucket, null, response, log);
+    }
+    return metadata.getBucket(request.bucketName, log, (err, data) => {
+        if (err) {
+            return responseXMLBody(errors.InternalError, null, response, log);
+        }
+        const isSystemXML = request.objectKey.endsWith('/system.xml');
+        const fileToBuild = isSystemXML ? data._capabilities?.VeeamSOSApi?.SystemInfo
+            : isSystemXML && !data._capabilities?.VeeamSOSApi?.CapacityInfo;
+        if (isSystemXML && !data._capabilities?.VeeamSOSApi?.SystemInfo) {
+            return responseXMLBody(errors.NoSuchKey, null, response, log);
+        } else if (!isSystemXML && !data._capabilities?.VeeamSOSApi?.CapacityInfo) {
+            return responseXMLBody(errors.NoSuchKey, null, response, log);
+        }
+        const savedModifiedDate = fileToBuild.LastModified;
+        delete fileToBuild.LastModified;
+        // Recompute file content to generate appropriate content-md5 header
+        const builder = new xml2js.Builder({
+            headless: true,
+        });
```

Comment:
this whole mapping of `system.xml` to `SystemInfo` and `CapacityInfo` otherwise to generate the XML is duplicate in both Head and Get, as well as list to some extend : should be factorized/de-duplicated (for get & head definitely, not sure how it would go for list...)

Reply:
see https://github.com/scality/cloudserver/pull/4942/commits/3941ed635c72ef5bf941e7a729d681a09403b525#diff-a23b06e1deac8b9332f76c3ae6723a32d6a49e7b7845e7f94f76e17190be0eb0R159

Reply:
Note: I squashed again to update the Arsenal version used and some linter issues. See https://github.com/scality/cloudserver/pull/4942/commits/067383064e138bb79da293d36b6189bd97ed3ee1